### PR TITLE
Add libbacktrace/1.0

### PR DESCRIPTION
**Pull request by klimkin**
**libbacktrace/1.0**

See #4262

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libbacktrace/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+
+
+class LibbacktraceConan(ConanFile):
+    name = "libbacktrace"
+    description = "A C library that may be linked into a C/C++ program to produce symbolic backtraces."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ianlancetaylor/libbacktrace"
+    license = "BSD-3-Clause"
```
Comment (klimkin) [inline recipes/libbacktrace/all/conanfile.py:9]: @ianlancetaylor Could you confirm the license for the library, please?

Comment (prince-chrismc) [reply recipes/libbacktrace/all/conanfile.py:9]: Seems the closet https://spdx.org/licenses/BSD-3-Clause.html

File: recipes/libbacktrace/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    @property
+    def scm(self):
+        source = {"type": "git"}
+        source.update(**self.conan_data["sources"][self.version])
+        return source
```
Comment (prince-chrismc) [inline recipes/libbacktrace/all/conanfile.py:32]: I do not think this method is supported by CCI 🤔

Comment (klimkin) [reply recipes/libbacktrace/all/conanfile.py:32]: libbacktrace doesn't provide releases.

Comment (klimkin) [reply recipes/libbacktrace/all/conanfile.py:32]: See https://github.com/ianlancetaylor/libbacktrace/issues/67

Comment (klimkin) [reply recipes/libbacktrace/all/conanfile.py:32]: It seems conandata.yml doesn't allow `revision` entry. Can this be relaxed? If allowed, we could use it to map the commit hashes to versions to enable reproducible builds. For example:
* 1.0-0: revision: xxxx
* 1.0-1: revision: yyyy

**Comment by klimkin** [comment]
Could you help to understand why the "apple-clang 11.0" and "clang 3.9, libstdc++" fails, please? Log files look good.

File: recipes/libbacktrace/all/conanfile.py [hunk 3]
```diff
@@ -49,6 +59,9 @@ def build(self):
     def package(self):
         self._autotools.install()
         self.copy("LICENSE", dst="licenses")
+        la = os.path.join(self.package_folder, "lib", "libbacktrace.la")
+        if os.path.exists(la):
+            os.unlink(la)
```
Comment (prince-chrismc) [inline recipes/libbacktrace/all/conanfile.py:64]: You can use this newer feature

https://github.com/conan-io/conan-center-index/commit/1b5321fa92d209203721febbb70504b97cc38dc6#diff-
7a002ca818a070c9265f9504710a98d337d6fc88fa1d907cb01b709f0b3975caR4-R66

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
Looks odd it printed `0xFFFFFFFF` for a pointer frame 🤔 No clue why it failed

File: recipes/libbacktrace/all/conanfile.py (hunk 3 above)
Comment (klimkin) [reply recipes/libbacktrace/all/conanfile.py]: Thanks!

File: recipes/libbacktrace/all/conanfile.py (hunk 1 above)
Comment (ianlancetaylor) [reply recipes/libbacktrace/all/conanfile.py:12]: Yes, this is a 3-clause BSD license.

**Comment by klimkin** [comment]
Looking at the logs, I don't see any errors, help?

**Comment by klimkin** [comment]
@prince-chrismc How can I understand the CI errors? I don't see anything failing in the logs...

**Comment by prince-chrismc** [comment]
I see this in the logs for apple-clang 11

```
/Applications/Xcode-11.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: .libs/libbacktrace.a(atomic.o) has no symbols
/Applications/Xcode-11.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: .libs/libbacktrace.a(atomic.o) has no symbols
/Applications/Xcode-11.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: /Users/jenkins/w/cci_PR-4431@2/.conan/data/libbacktrace/1.0/_/_/package/0aa7be2d0914b6f3128105e305c90b9cb0cae86d/lib/libbacktrace.a(atomic.o) has no symbols
```

For 3.9 and 4, I see back command line arguments.

***

I suggest you block `libc++` ... we'll see if that's the problem 🤞 we get a build to pass

There's a new backend for the CI system released and we'll get more logs if we try again

Not every single configuration needs to be supported when added a recipe it's okay if we cut `libc++`

**Comment by prince-chrismc** [comment]
https://c3i.jfrog.io/c3i/misc/logs/pr/4431/18/libbacktrace/1.0

You need to retrigger CI

File: recipes/libbacktrace/all/conanfile.py (hunk 2 above)
Comment (jgsogo) [reply recipes/libbacktrace/all/conanfile.py:50]: There is a version schema for libraries that doesn't provide explicit versions: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

OTH, I think this is breaking the current CI process. When I run it locally I get an error

```
ERROR: libbacktrace/1.0: Cannot load recipe.
'property' object has no attribute 'update' 
```

Indeed, every _public_ function or member in the `ConanFile` is reserved to Conan, and `scm` is an attribute we are already using.

**Review by jgsogo — CHANGES_REQUESTED** [review]
Please, read the comment above, I'll keep having a look at the unexpected error.

**Comment by jgsogo** [comment]
I see it doesn't fail in the docker image (`gcc9` - Python 3.7.4), but it does in some other Python installations (newer versions, Macos). I'll try to improve the current CI to provide information about this last step where the PR is failing.

File: recipes/libbacktrace/all/conanfile.py (hunk 2 above)
Comment (klimkin) [reply recipes/libbacktrace/all/conanfile.py]: Thanks!
https://github.com/conan-io/conan-center-index/pull/4431/commits/d24ad500c1d844f27ae38e5fd4d5706c30b090ac

**Review by jgsogo — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Pull request merged**